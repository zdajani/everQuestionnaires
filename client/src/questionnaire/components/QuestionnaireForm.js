import React from 'react';
import PropTypes from 'prop-types';
import { Field, FieldArray, reduxForm } from 'redux-form';
import { validate } from '../validations';
import renderField from '../../commonComponents/Field';
import QuestionFieldArray from '../../question/components/QuestionFieldArray';

export const QuestionnaireForm = props => {  
  const { handleSubmit, pristine, reset, submitting } = props
  return (
    <div className="container">
    <form onSubmit={handleSubmit} className="container">
      <h2 className="text-center">Create Questionnaire</h2>
      <Field
        name="title"
        type="text"
        component={renderField}
        label="Questionnaire Title"
      />
      <FieldArray name="questions" component={QuestionFieldArray} />
      <div>
        <button type="button" className="btn btn-outline-danger btn-lg btn-block" disabled={pristine || submitting} onClick={reset}>
          Clear Values
        </button>
        <button className="btn btn-outline-success btn-lg btn-lg btn-block" type="submit" disabled={submitting}>Submit</button>
      </div>
    </form>
  </div>
  );
};

QuestionnaireForm.propTypes = {
  pristine: PropTypes.bool,
	handleSubmit: PropTypes.func.isRequired,
	reset: PropTypes.func.isRequired,
	submitting: PropTypes.bool.isRequired
}

export default reduxForm({
  form: 'QuestionnaireForm',
  validate
})(QuestionnaireForm);