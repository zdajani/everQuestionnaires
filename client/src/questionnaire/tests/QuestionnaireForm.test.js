import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import QuestionnaireForm from '../components/QuestionnaireForm';

import { reducer as formReducer } from 'redux-form'
import { createStore, combineReducers } from 'redux'
import { Provider } from 'react-redux'

const setup = () => {
  const store = createStore(combineReducers({ form: formReducer }))
  const resetSpy = jest.fn();
  const handleSubmitSpy = jest.fn();
  
  const mountedComponent = mount(
    <Provider store={store}>
      <QuestionnaireForm 
        handleSubmit={handleSubmitSpy}
        reset={resetSpy}
        submitting={false}  />
    </Provider>);
    
  return {
    mountedComponent,
    resetSpy,
    handleSubmitSpy
  };
}

describe('QuestionnaireForm', () => {
  const { mountedComponent, handleSubmitSpy } = setup()
  it('renders self with props', () => {  
    expect(toJson(mountedComponent)).toMatchSnapshot();
  });
  
  it('renders calls handleSubmit on submit button', () => {  
    mountedComponent.find('button').first().simulate('submit');
    expect(handleSubmitSpy.mock.calls.length).toBe(1)
  });
    
});