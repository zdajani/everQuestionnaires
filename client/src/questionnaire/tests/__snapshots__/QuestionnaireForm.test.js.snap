// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionnaireForm renders self with props 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ReduxForm
    handleSubmit={[Function]}
    reset={[Function]}
    submitting={false}
  >
    <Connect(Form(Component))
      destroyOnUnmount={true}
      enableReinitialize={false}
      forceUnregisterOnUnmount={false}
      form="QuestionnaireForm"
      getFormState={[Function]}
      handleSubmit={[Function]}
      keepDirtyOnReinitialize={false}
      persistentSubmitErrors={false}
      pure={true}
      reset={[Function]}
      shouldAsyncValidate={[Function]}
      shouldValidate={[Function]}
      submitting={false}
      touchOnBlur={true}
      touchOnChange={false}
      validate={[Function]}
    >
      <Form(Component)
        anyTouched={false}
        array={
          Object {
            "insert": [Function],
            "move": [Function],
            "pop": [Function],
            "push": [Function],
            "remove": [Function],
            "removeAll": [Function],
            "shift": [Function],
            "splice": [Function],
            "swap": [Function],
            "unshift": [Function],
          }
        }
        arrayInsert={[Function]}
        arrayMove={[Function]}
        arrayPop={[Function]}
        arrayPush={[Function]}
        arrayRemove={[Function]}
        arrayRemoveAll={[Function]}
        arrayShift={[Function]}
        arraySplice={[Function]}
        arraySwap={[Function]}
        arrayUnshift={[Function]}
        asyncValidating={false}
        autofill={[Function]}
        blur={[Function]}
        change={[Function]}
        clearAsyncError={[Function]}
        clearSubmit={[Function]}
        clearSubmitErrors={[Function]}
        destroy={[Function]}
        destroyOnUnmount={true}
        dirty={false}
        dispatch={[Function]}
        enableReinitialize={false}
        focus={[Function]}
        forceUnregisterOnUnmount={false}
        form="QuestionnaireForm"
        getFormState={[Function]}
        handleSubmit={[Function]}
        initialize={[Function]}
        initialized={false}
        invalid={true}
        keepDirtyOnReinitialize={false}
        persistentSubmitErrors={false}
        pristine={true}
        pure={true}
        registerField={[Function]}
        registeredFields={
          Object {
            "questions": Object {
              "count": 1,
              "name": "questions",
              "type": "FieldArray",
            },
            "title": Object {
              "count": 1,
              "name": "title",
              "type": "Field",
            },
          }
        }
        reset={[Function]}
        setSubmitFailed={[Function]}
        setSubmitSucceeded={[Function]}
        shouldAsyncValidate={[Function]}
        shouldValidate={[Function]}
        startAsyncValidation={[Function]}
        startSubmit={[Function]}
        stopAsyncValidation={[Function]}
        stopSubmit={[Function]}
        submit={[Function]}
        submitFailed={false}
        submitSucceeded={false}
        submitting={false}
        syncErrors={
          Object {
            "questions": Object {
              "_error": "At least one question must be entered",
            },
            "title": "Required",
          }
        }
        syncWarnings={Object {}}
        touch={[Function]}
        touchOnBlur={true}
        touchOnChange={false}
        unregisterField={[Function]}
        untouch={[Function]}
        updateSyncErrors={[Function]}
        updateSyncWarnings={[Function]}
        valid={false}
        validExceptSubmit={false}
        validate={[Function]}
        values={Object {}}
      >
        <Component
          anyTouched={false}
          array={
            Object {
              "insert": [Function],
              "move": [Function],
              "pop": [Function],
              "push": [Function],
              "remove": [Function],
              "removeAll": [Function],
              "shift": [Function],
              "splice": [Function],
              "swap": [Function],
              "unshift": [Function],
            }
          }
          asyncValidate={[Function]}
          asyncValidating={false}
          autofill={[Function]}
          blur={[Function]}
          change={[Function]}
          clearAsyncError={[Function]}
          clearSubmit={[Function]}
          clearSubmitErrors={[Function]}
          destroy={[Function]}
          dirty={false}
          dispatch={[Function]}
          form="QuestionnaireForm"
          handleSubmit={[Function]}
          initialize={[Function]}
          initialized={false}
          invalid={true}
          pristine={true}
          pure={true}
          reset={[Function]}
          submit={[Function]}
          submitFailed={false}
          submitSucceeded={false}
          submitting={false}
          touch={[Function]}
          untouch={[Function]}
          valid={false}
          validate={[Function]}
        >
          <div
            className="container"
          >
            <form
              className="container"
              onSubmit={[Function]}
            >
              <h2
                className="text-center"
              >
                Create Questionnaire
              </h2>
              <Field
                component={[Function]}
                label="Questionnaire Title"
                name="title"
                type="text"
              >
                <Connect(ConnectedField)
                  _reduxForm={
                    Object {
                      "anyTouched": false,
                      "array": Object {
                        "insert": [Function],
                        "move": [Function],
                        "pop": [Function],
                        "push": [Function],
                        "remove": [Function],
                        "removeAll": [Function],
                        "shift": [Function],
                        "splice": [Function],
                        "swap": [Function],
                        "unshift": [Function],
                      },
                      "arrayInsert": [Function],
                      "arrayMove": [Function],
                      "arrayPop": [Function],
                      "arrayPush": [Function],
                      "arrayRemove": [Function],
                      "arrayRemoveAll": [Function],
                      "arrayShift": [Function],
                      "arraySplice": [Function],
                      "arraySwap": [Function],
                      "arrayUnshift": [Function],
                      "asyncErrors": undefined,
                      "asyncValidate": [Function],
                      "asyncValidating": false,
                      "autofill": [Function],
                      "blur": [Function],
                      "change": [Function],
                      "clearAsyncError": [Function],
                      "clearSubmit": [Function],
                      "clearSubmitErrors": [Function],
                      "destroy": [Function],
                      "destroyOnUnmount": true,
                      "dirty": false,
                      "dispatch": [Function],
                      "enableReinitialize": false,
                      "error": undefined,
                      "focus": [Function],
                      "forceUnregisterOnUnmount": false,
                      "form": "QuestionnaireForm",
                      "getFormState": [Function],
                      "getValues": [Function],
                      "handleSubmit": [Function],
                      "initialValues": undefined,
                      "initialize": [Function],
                      "initialized": false,
                      "invalid": true,
                      "keepDirtyOnReinitialize": false,
                      "persistentSubmitErrors": false,
                      "pristine": true,
                      "pure": true,
                      "register": [Function],
                      "registerField": [Function],
                      "registerInnerOnSubmit": [Function],
                      "registeredFields": Object {
                        "questions": Object {
                          "count": 1,
                          "name": "questions",
                          "type": "FieldArray",
                        },
                        "title": Object {
                          "count": 1,
                          "name": "title",
                          "type": "Field",
                        },
                      },
                      "reset": [Function],
                      "sectionPrefix": undefined,
                      "setSubmitFailed": [Function],
                      "setSubmitSucceeded": [Function],
                      "shouldAsyncValidate": [Function],
                      "shouldValidate": [Function],
                      "startAsyncValidation": [Function],
                      "startSubmit": [Function],
                      "stopAsyncValidation": [Function],
                      "stopSubmit": [Function],
                      "submit": [Function],
                      "submitFailed": false,
                      "submitSucceeded": false,
                      "submitting": false,
                      "syncErrors": Object {
                        "questions": Object {
                          "_error": "At least one question must be entered",
                        },
                        "title": "Required",
                      },
                      "syncWarnings": Object {},
                      "touch": [Function],
                      "touchOnBlur": true,
                      "touchOnChange": false,
                      "triggerSubmit": undefined,
                      "unregister": [Function],
                      "unregisterField": [Function],
                      "untouch": [Function],
                      "updateSyncErrors": [Function],
                      "updateSyncWarnings": [Function],
                      "valid": false,
                      "validExceptSubmit": false,
                      "validate": [Function],
                      "values": Object {},
                      "warning": undefined,
                    }
                  }
                  component={[Function]}
                  label="Questionnaire Title"
                  name="title"
                  normalize={[Function]}
                  type="text"
                >
                  <ConnectedField
                    _reduxForm={
                      Object {
                        "anyTouched": false,
                        "array": Object {
                          "insert": [Function],
                          "move": [Function],
                          "pop": [Function],
                          "push": [Function],
                          "remove": [Function],
                          "removeAll": [Function],
                          "shift": [Function],
                          "splice": [Function],
                          "swap": [Function],
                          "unshift": [Function],
                        },
                        "arrayInsert": [Function],
                        "arrayMove": [Function],
                        "arrayPop": [Function],
                        "arrayPush": [Function],
                        "arrayRemove": [Function],
                        "arrayRemoveAll": [Function],
                        "arrayShift": [Function],
                        "arraySplice": [Function],
                        "arraySwap": [Function],
                        "arrayUnshift": [Function],
                        "asyncErrors": undefined,
                        "asyncValidate": [Function],
                        "asyncValidating": false,
                        "autofill": [Function],
                        "blur": [Function],
                        "change": [Function],
                        "clearAsyncError": [Function],
                        "clearSubmit": [Function],
                        "clearSubmitErrors": [Function],
                        "destroy": [Function],
                        "destroyOnUnmount": true,
                        "dirty": false,
                        "dispatch": [Function],
                        "enableReinitialize": false,
                        "error": undefined,
                        "focus": [Function],
                        "forceUnregisterOnUnmount": false,
                        "form": "QuestionnaireForm",
                        "getFormState": [Function],
                        "getValues": [Function],
                        "handleSubmit": [Function],
                        "initialValues": undefined,
                        "initialize": [Function],
                        "initialized": false,
                        "invalid": true,
                        "keepDirtyOnReinitialize": false,
                        "persistentSubmitErrors": false,
                        "pristine": true,
                        "pure": true,
                        "register": [Function],
                        "registerField": [Function],
                        "registerInnerOnSubmit": [Function],
                        "registeredFields": Object {
                          "questions": Object {
                            "count": 1,
                            "name": "questions",
                            "type": "FieldArray",
                          },
                          "title": Object {
                            "count": 1,
                            "name": "title",
                            "type": "Field",
                          },
                        },
                        "reset": [Function],
                        "sectionPrefix": undefined,
                        "setSubmitFailed": [Function],
                        "setSubmitSucceeded": [Function],
                        "shouldAsyncValidate": [Function],
                        "shouldValidate": [Function],
                        "startAsyncValidation": [Function],
                        "startSubmit": [Function],
                        "stopAsyncValidation": [Function],
                        "stopSubmit": [Function],
                        "submit": [Function],
                        "submitFailed": false,
                        "submitSucceeded": false,
                        "submitting": false,
                        "syncErrors": Object {
                          "questions": Object {
                            "_error": "At least one question must be entered",
                          },
                          "title": "Required",
                        },
                        "syncWarnings": Object {},
                        "touch": [Function],
                        "touchOnBlur": true,
                        "touchOnChange": false,
                        "triggerSubmit": undefined,
                        "unregister": [Function],
                        "unregisterField": [Function],
                        "untouch": [Function],
                        "updateSyncErrors": [Function],
                        "updateSyncWarnings": [Function],
                        "valid": false,
                        "validExceptSubmit": false,
                        "validate": [Function],
                        "values": Object {},
                        "warning": undefined,
                      }
                    }
                    asyncValidating={false}
                    component={[Function]}
                    dirty={false}
                    dispatch={[Function]}
                    label="Questionnaire Title"
                    name="title"
                    normalize={[Function]}
                    pristine={true}
                    syncError="Required"
                    type="text"
                  >
                    <renderField
                      input={
                        Object {
                          "name": "title",
                          "onBlur": [Function],
                          "onChange": [Function],
                          "onDragStart": [Function],
                          "onDrop": [Function],
                          "onFocus": [Function],
                          "value": "",
                        }
                      }
                      label="Questionnaire Title"
                      meta={
                        Object {
                          "active": false,
                          "asyncValidating": false,
                          "autofilled": false,
                          "dirty": false,
                          "dispatch": [Function],
                          "error": "Required",
                          "form": "QuestionnaireForm",
                          "initial": undefined,
                          "invalid": true,
                          "pristine": true,
                          "submitFailed": false,
                          "submitting": false,
                          "touched": false,
                          "valid": false,
                          "visited": false,
                          "warning": undefined,
                        }
                      }
                      type="text"
                    >
                      <div
                        className="form-group "
                      >
                        <label
                          className="form-control-label"
                        >
                          Questionnaire Title
                        </label>
                        <input
                          className="form-control"
                          name="title"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onDragStart={[Function]}
                          onDrop={[Function]}
                          onFocus={[Function]}
                          placeholder="Questionnaire Title"
                          type="text"
                          value=""
                        />
                        <div
                          className="text-help"
                        />
                      </div>
                    </renderField>
                  </ConnectedField>
                </Connect(ConnectedField)>
              </Field>
              <FieldArray
                component={[Function]}
                name="questions"
              >
                <Connect(ConnectedFieldArray)
                  _reduxForm={
                    Object {
                      "anyTouched": false,
                      "array": Object {
                        "insert": [Function],
                        "move": [Function],
                        "pop": [Function],
                        "push": [Function],
                        "remove": [Function],
                        "removeAll": [Function],
                        "shift": [Function],
                        "splice": [Function],
                        "swap": [Function],
                        "unshift": [Function],
                      },
                      "arrayInsert": [Function],
                      "arrayMove": [Function],
                      "arrayPop": [Function],
                      "arrayPush": [Function],
                      "arrayRemove": [Function],
                      "arrayRemoveAll": [Function],
                      "arrayShift": [Function],
                      "arraySplice": [Function],
                      "arraySwap": [Function],
                      "arrayUnshift": [Function],
                      "asyncErrors": undefined,
                      "asyncValidate": [Function],
                      "asyncValidating": false,
                      "autofill": [Function],
                      "blur": [Function],
                      "change": [Function],
                      "clearAsyncError": [Function],
                      "clearSubmit": [Function],
                      "clearSubmitErrors": [Function],
                      "destroy": [Function],
                      "destroyOnUnmount": true,
                      "dirty": false,
                      "dispatch": [Function],
                      "enableReinitialize": false,
                      "error": undefined,
                      "focus": [Function],
                      "forceUnregisterOnUnmount": false,
                      "form": "QuestionnaireForm",
                      "getFormState": [Function],
                      "getValues": [Function],
                      "handleSubmit": [Function],
                      "initialValues": undefined,
                      "initialize": [Function],
                      "initialized": false,
                      "invalid": true,
                      "keepDirtyOnReinitialize": false,
                      "persistentSubmitErrors": false,
                      "pristine": true,
                      "pure": true,
                      "register": [Function],
                      "registerField": [Function],
                      "registerInnerOnSubmit": [Function],
                      "registeredFields": Object {
                        "questions": Object {
                          "count": 1,
                          "name": "questions",
                          "type": "FieldArray",
                        },
                        "title": Object {
                          "count": 1,
                          "name": "title",
                          "type": "Field",
                        },
                      },
                      "reset": [Function],
                      "sectionPrefix": undefined,
                      "setSubmitFailed": [Function],
                      "setSubmitSucceeded": [Function],
                      "shouldAsyncValidate": [Function],
                      "shouldValidate": [Function],
                      "startAsyncValidation": [Function],
                      "startSubmit": [Function],
                      "stopAsyncValidation": [Function],
                      "stopSubmit": [Function],
                      "submit": [Function],
                      "submitFailed": false,
                      "submitSucceeded": false,
                      "submitting": false,
                      "syncErrors": Object {
                        "questions": Object {
                          "_error": "At least one question must be entered",
                        },
                        "title": "Required",
                      },
                      "syncWarnings": Object {},
                      "touch": [Function],
                      "touchOnBlur": true,
                      "touchOnChange": false,
                      "triggerSubmit": undefined,
                      "unregister": [Function],
                      "unregisterField": [Function],
                      "untouch": [Function],
                      "updateSyncErrors": [Function],
                      "updateSyncWarnings": [Function],
                      "valid": false,
                      "validExceptSubmit": false,
                      "validate": [Function],
                      "values": Object {},
                      "warning": undefined,
                    }
                  }
                  component={[Function]}
                  name="questions"
                >
                  <ConnectedFieldArray
                    _reduxForm={
                      Object {
                        "anyTouched": false,
                        "array": Object {
                          "insert": [Function],
                          "move": [Function],
                          "pop": [Function],
                          "push": [Function],
                          "remove": [Function],
                          "removeAll": [Function],
                          "shift": [Function],
                          "splice": [Function],
                          "swap": [Function],
                          "unshift": [Function],
                        },
                        "arrayInsert": [Function],
                        "arrayMove": [Function],
                        "arrayPop": [Function],
                        "arrayPush": [Function],
                        "arrayRemove": [Function],
                        "arrayRemoveAll": [Function],
                        "arrayShift": [Function],
                        "arraySplice": [Function],
                        "arraySwap": [Function],
                        "arrayUnshift": [Function],
                        "asyncErrors": undefined,
                        "asyncValidate": [Function],
                        "asyncValidating": false,
                        "autofill": [Function],
                        "blur": [Function],
                        "change": [Function],
                        "clearAsyncError": [Function],
                        "clearSubmit": [Function],
                        "clearSubmitErrors": [Function],
                        "destroy": [Function],
                        "destroyOnUnmount": true,
                        "dirty": false,
                        "dispatch": [Function],
                        "enableReinitialize": false,
                        "error": undefined,
                        "focus": [Function],
                        "forceUnregisterOnUnmount": false,
                        "form": "QuestionnaireForm",
                        "getFormState": [Function],
                        "getValues": [Function],
                        "handleSubmit": [Function],
                        "initialValues": undefined,
                        "initialize": [Function],
                        "initialized": false,
                        "invalid": true,
                        "keepDirtyOnReinitialize": false,
                        "persistentSubmitErrors": false,
                        "pristine": true,
                        "pure": true,
                        "register": [Function],
                        "registerField": [Function],
                        "registerInnerOnSubmit": [Function],
                        "registeredFields": Object {
                          "questions": Object {
                            "count": 1,
                            "name": "questions",
                            "type": "FieldArray",
                          },
                          "title": Object {
                            "count": 1,
                            "name": "title",
                            "type": "Field",
                          },
                        },
                        "reset": [Function],
                        "sectionPrefix": undefined,
                        "setSubmitFailed": [Function],
                        "setSubmitSucceeded": [Function],
                        "shouldAsyncValidate": [Function],
                        "shouldValidate": [Function],
                        "startAsyncValidation": [Function],
                        "startSubmit": [Function],
                        "stopAsyncValidation": [Function],
                        "stopSubmit": [Function],
                        "submit": [Function],
                        "submitFailed": false,
                        "submitSucceeded": false,
                        "submitting": false,
                        "syncErrors": Object {
                          "questions": Object {
                            "_error": "At least one question must be entered",
                          },
                          "title": "Required",
                        },
                        "syncWarnings": Object {},
                        "touch": [Function],
                        "touchOnBlur": true,
                        "touchOnChange": false,
                        "triggerSubmit": undefined,
                        "unregister": [Function],
                        "unregisterField": [Function],
                        "untouch": [Function],
                        "updateSyncErrors": [Function],
                        "updateSyncWarnings": [Function],
                        "valid": false,
                        "validExceptSubmit": false,
                        "validate": [Function],
                        "values": Object {},
                        "warning": undefined,
                      }
                    }
                    arrayInsert={[Function]}
                    arrayMove={[Function]}
                    arrayPop={[Function]}
                    arrayPush={[Function]}
                    arrayRemove={[Function]}
                    arrayRemoveAll={[Function]}
                    arrayShift={[Function]}
                    arraySplice={[Function]}
                    arraySwap={[Function]}
                    arrayUnshift={[Function]}
                    component={[Function]}
                    dirty={false}
                    length={0}
                    name="questions"
                    pristine={true}
                    rerenderOnEveryChange={false}
                    syncError="At least one question must be entered"
                  >
                    <QuestionFieldArray
                      fields={
                        Object {
                          "_isFieldArray": true,
                          "forEach": [Function],
                          "get": [Function],
                          "getAll": [Function],
                          "insert": [Function],
                          "length": 0,
                          "map": [Function],
                          "move": [Function],
                          "name": "questions",
                          "pop": [Function],
                          "push": [Function],
                          "reduce": [Function],
                          "remove": [Function],
                          "removeAll": [Function],
                          "shift": [Function],
                          "swap": [Function],
                          "unshift": [Function],
                        }
                      }
                      meta={
                        Object {
                          "dirty": false,
                          "error": "At least one question must be entered",
                          "form": "QuestionnaireForm",
                          "invalid": true,
                          "pristine": true,
                          "submitFailed": undefined,
                          "submitting": undefined,
                          "valid": false,
                          "warning": undefined,
                        }
                      }
                    >
                      <div>
                        <div
                          className="form-group"
                        >
                          <button
                            className="btn btn-outline-info btn-lg"
                            onClick={[Function]}
                            type="button"
                          >
                            Add Question
                          </button>
                        </div>
                      </div>
                    </QuestionFieldArray>
                  </ConnectedFieldArray>
                </Connect(ConnectedFieldArray)>
              </FieldArray>
              <div>
                <button
                  className="btn btn-outline-danger btn-lg btn-block"
                  disabled={true}
                  onClick={[Function]}
                  type="button"
                >
                  Clear Values
                </button>
                <button
                  className="btn btn-outline-success btn-lg btn-lg btn-block"
                  disabled={false}
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </div>
        </Component>
      </Form(Component)>
    </Connect(Form(Component))>
  </ReduxForm>
</Provider>
`;
